Linter doesn't like the PKCE code in handlers.go

I'm seeing this:

~~~~
curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | BINARY=golangci-lint bash -s -- v1.31.0
golangci/golangci-lint info checking GitHub for tag 'v1.31.0'
golangci/golangci-lint info found version: 1.31.0 for v1.31.0/linux/amd64
golangci/golangci-lint info installed ./bin/golangci-lint
bin/golangci-lint run
Error: server/handlers.go:837:2: ifElseChain: rewrite if-else to switch statement (gocritic)
	if providedCodeVerifier != "" && codeChallengeFromStorage != "" {
	^
make: *** [lint] Error 1
Error: Makefile:70: recipe for target 'lint' failed
Error: Process completed with exit code 2.
~~~~

--- a/connector/ldap/ldap.go
+++ b/connector/ldap/ldap.go
@@ -62,6 +62,7 @@ type UserMatcher struct {
 	GroupAttr string `json:"groupAttr"`
 }
 
+// Config holds configuration options for LDAP logins.
 type Config struct {
 	// The host and optional port of the LDAP server. If port isn't supplied, it will be
 	// guessed based on the TLS configuration. 389 or 636.

--- a/server/handlers.go
+++ b/server/handlers.go
@@ -820,7 +820,8 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 	codeChallengeFromStorage := authCode.PKCE.CodeChallenge
 	providedCodeVerifier := r.PostFormValue("code_verifier")
 
-	if providedCodeVerifier != "" && codeChallengeFromStorage != "" {
+	switch {
+	case providedCodeVerifier != "" && codeChallengeFromStorage != "":
 		calculatedCodeChallenge, err := s.calculateCodeChallenge(providedCodeVerifier, authCode.PKCE.CodeChallengeMethod)
 		if err != nil {
 			s.logger.Error(err)
@@ -831,11 +832,11 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 			s.tokenErrHelper(w, errInvalidGrant, "Invalid code_verifier.", http.StatusBadRequest)
 			return
 		}
-	} else if providedCodeVerifier != "" {
+	case providedCodeVerifier != "":
 		// Received no code_challenge on /auth, but a code_verifier on /token
 		s.tokenErrHelper(w, errInvalidRequest, "No PKCE flow started. Cannot check code_verifier.", http.StatusBadRequest)
 		return
-	} else if codeChallengeFromStorage != "" {
+	case codeChallengeFromStorage != "":
 		// Received PKCE request on /auth, but no code_verifier on /token
 		s.tokenErrHelper(w, errInvalidGrant, "Expecting parameter code_verifier in PKCE flow.", http.StatusBadRequest)
 		return

